modules/log/module.go

```diff
@@ -58,7 +64,7 @@ func New(logger *zap.Logger) *Module {
 }
 
 func (m *Module) Apply(r *domain.Resource) (domain.ResourceStatus, error) {
-	switch r.Configs[levelConfigString].(Level) {
+	switch Level(r.Configs[levelConfigString].(string)) {
```

Comment:
will it be necessary to write in more defensive way when typecasting it?
```
levelCfg, ok := r.Configs[levelConfigString].(string)
```

Reply:
Sounds good, let me do this.

Reply:
Configs are something that we would have a lot of in all modules, so instead of actually doing it in multiple times, I think it is better to create a config struct and unmarshal the map to it. Pushed a new commit for same.